client/src/components/molecules/DonationItemForm.tsx

```diff
@@ -0,0 +1,307 @@
+        const errors = [];
+        if (updateIsConditionErrorneous(condition)) {
+            errors.push("Item Condition");
+        }
+        if (updateIsQuantityErrorneous(quantity)) {
+            errors.push("Quantity (number of this specific item)");
+        }
+        if (errors.length === 0) {
+            setFormError("");
+            return "";
+        }
+        let errorString = "Please enter valid";
+        errors.forEach((error, idx) => {
+            if (idx === errors.length - 1 && errors.length > 1) {
+                errorString = errorString.concat(" and");
+            }
+            errorString = errorString.concat(" " + error);
+        });
+        errorString = errorString.concat(".");
+        setFormError(errorString);
+
+        return errorString;
+    };
+
+    const onConditionChange = (newCondition: ItemCondition) => {
+        setCondition(newCondition);
+        setIsConditionDropdownOpen(false);
+        updateFormError(newCondition, quantity);
+    };
+
+    const onAgeChange = (newAge: number) => {
+        setAge(newAge);
+        setIsAgeDropdownOpen(false);
+    };
+
+    const onQuantityChange = (newQuantity: number) => {
+        setQuantity(newQuantity);
+        updateFormError(condition, newQuantity);
+    };
+
+    const onDescriptionChange = (newDescription: string) => {
+        setDescription(newDescription);
+    };
+
+    const onSave = () => {
+        const nameError = updateNameError(name);
+        const formError = updateFormError(condition, quantity);
+
+        if (nameError.length === 0 && formError.length === 0) {
+            const itemRequestGroup = props.requestGroups.find((requestGroup) => requestGroup.name === name);
+
+            props.onSave({
+                age,
+                condition: condition ?? undefined,
+                description,
+                name,
+                quantity,
+                requestGroup: itemRequestGroup ?? null
+            });
+        }
+    };
+
+    return (
+        <div className="donation-item-form">
+            <FormItem
+                className="item-name-field"
+                formItemName="What item do you wish to donate?"
+                errorString={nameError}
+                isDisabled={false}
+                showErrorIcon={false}
+                inputComponent={
+                    <SearchableDropdown
+                        selectedItem={name}
+                        searchString={nameInput}
+                        dropdownItems={props.requestGroups.reduce(
+                            (acc, requestGroup) => (requestGroup.name ? acc.concat(requestGroup.name) : acc),
+                            [] as Array<string>
+                        )}
+                        initialText={nameInput}
+                        isDisabled={false}
+                        isErroneous={nameError.length !== 0}
+                        actionOption={
+                            nameInput.length === 0 ? null : (
```

Comment:
I would also add a check to make sure that nameInput is not part of props.requestGroups otherwise we would see
ex. "Bassinet - enter your own item" even though Bassinet is already an existing item and is chosen.

Just my 2 cents

Reply:
great suggestion! ill do that